hoomd/hpmc/pytest/test_shape.py

```diff
@@ -0,0 +1,109 @@
+import hoomd
+import hoomd.hpmc
+from hoomd.hpmc import _hpmc
+import numpy as np
+
+
+def test_dict_conversion(shape_dict_conversion_args):
+    for shape_params, args_list in shape_dict_conversion_args():
+        for args in args_list:
+            test_shape = shape_params(args)
+            test_dict = test_shape.asDict()
+            for key, val in list(args.items()):
+                if isinstance(test_dict[key], list) and len(test_dict[key]) > 0 \
+                and key != 'shapes':
+                    np.testing.assert_allclose(test_dict[key], val)
+                elif key != 'shapes':
+                    assert test_dict[key] == val
+
+
+def test_default_args(integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        mc = shape_integrator(23456)
+        mc.shape['A'] = dict()
+        for key, val in list(mc.shape["A"].items()):
+            assert mc.shape["A"][key] == val
+
+
+def test_shape_params(integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        mc = shape_integrator(23456)
+        for args in valid_args:
+            mc.shape["A"] = args
+            for key, val in list(args.items()):
+                if key == "vertices":
+                    np.testing.assert_allclose(mc.shape["A"][key], val)
+                else:
+                    assert mc.shape["A"][key] == val
+        for args in invalid_args:
+            tf = False
+            try:
+                mc.shape["A"] = args
+            except:
+                tf = True
+            assert tf
+
+
+def test_shape_attached(dummy_simulation_factory, integrator_args):
+    for shape_integrator, valid_args, invalid_args in integrator_args():
+        mc = shape_integrator(23456)
+        for args in valid_args:
+            mc.shape["A"] = args
+            sim = dummy_simulation_factory()
+            sim.operations.add(mc)
+            sim.operations.schedule()
+            for key, val in list(args.items()):
+                if isinstance(mc.shape["A"][key], list) and len(mc.shape["A"][key]) > 0 \
+                and key != 'shapes':
```

Comment:
Why don't we do any checks for `shapes`. This is for the unions right? Do we currently have a problem converting back to python?